database/manager/manager.go

```diff
@@ -0,0 +1,238 @@
+// (c) 2019-2020, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package manager
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/database/leveldb"
+	"github.com/ava-labs/avalanchego/database/memdb"
+	"github.com/ava-labs/avalanchego/database/meterdb"
+	"github.com/ava-labs/avalanchego/database/prefixdb"
+	"github.com/ava-labs/avalanchego/utils"
+	"github.com/ava-labs/avalanchego/utils/wrappers"
+	"github.com/ava-labs/avalanchego/version"
+)
+
+var (
+	errNonSortedAndUniqueDBs = errors.New("managed databases were not sorted and unique")
+)
+
+type Manager interface {
+	// Current returns the database with the current database version
+	Current() *SemanticDatabase
+	// Last returns the last database prior to the current database or an error if none exists
+	Last() (*SemanticDatabase, bool)
```

Comment:
Nit: Can we change `Last()` to `Previous()`? I think `Last` could be misconstrued as `Current`

Reply:
Good call